core/runtime/ops.rs

```diff
@@ -215,140 +215,202 @@ mod tests {
   use crate::JsRuntime;
   use crate::RuntimeOptions;
   use deno_ops::op2;
+  use std::cell::Cell;
 
   crate::extension!(
     testing,
     ops = [
+      op_test_fail,
       op_test_add,
       op_test_add_option,
+      op_test_result_void_switch,
       op_test_result_void_ok,
       op_test_result_void_err,
       op_test_result_primitive_ok,
       op_test_result_primitive_err
     ]
   );
 
+  thread_local! {
+    static FAIL: Cell<bool> = Cell::new(false)
+  }
+
+  #[op2(core, fast)]
+  pub fn op_test_fail() {
+    FAIL.with(|b| {
+      println!("fail");
+      b.set(true)
+    })
+  }
+
   /// Run a test for a single op.
-  fn run_test(
+  fn run_test2(
```

Comment:
This works, but `%OptimizeFunctionOnNextCall` would be a better approach. I'll address this in a future patch.